se/se_epub_build.py

```diff
@@ -234,18 +234,22 @@ def build(self, metadata_xhtml: str, metadata_tree: se.easy_xml.EasyXmlTree, run
 					force_convert = False
 					for selector in selectors:
 						try:
-							sel = lxml.cssselect.CSSSelector(selector, translator="xhtml", namespaces=se.XHTML_NAMESPACES)
-
 							# Add classes to elements that match any of our selectors to simplify. For example, if we select :first-child, add a "first-child" class to all elements that match that.
 							for selector_to_simplify in se.SELECTORS_TO_SIMPLIFY:
-								if selector_to_simplify in selector:
-									selector_to_simplify = selector_to_simplify.replace(":", "")
+								while selector_to_simplify in selector:
```

Comment:
In the case of something like p:first-child:last-child when there are multiple 'p' elements, this algorithm marks the first 'p' element as class="first-child" and does not mark the last element. The first time through the loop it selects 'p:first-child' which succeeds and the second time through it tries to select 'p:first-child:last-child" which fails. I would think we would not want to mark either 'p' element because the selector is not actually active?

Reply:
We rewrite selector to replace the pseudoclass with a “real” class on line 245, so the second pass should actually try to read `p.first-child:last-child` which should succeed. Although it’s entirely possible that I’ve misinterpreted how Python while loops work. It’s fair to say that I haven’t tested this path properly, I was more interested in the `p:first-child + p` scenario.